rag_agent.py

```diff
@@ -71,74 +66,72 @@ def setup_vectorstore(self, file_paths):
                     loader = PyMuPDFLoader(file_path)
                 else:
                     loader = TextLoader(file_path)
-    
+
                 documents = loader.load()
                 all_documents.extend(documents)
 
+        # Using HuggingFace embeddings
         embeddings = HuggingFaceEmbeddings()
         vector_store = FAISS.from_documents(all_documents, embeddings)
         retriever = vector_store.as_retriever()
-
         return retriever
 
-    def get_relevant_document(self, query, retriever, threshold=0.5):
+
+    def is_coding_query(self, question):
         """
-        Check if the query is related to the document by using the retriever.
-    
+        Check if the question is coding-related by looking for specific keywords.
+        
         Args:
-            query (str): The user query.
-            retriever: The document retriever.
-            threshold (float): The confidence threshold to decide
-                               if a query is related to the document.
-    
+            question (str): The user query.
+        
         Returns:
-            result (dict): Retrieved document information and similarity score.
+            bool: True if the query is coding-related, False otherwise.
         """
-        self.retriever = self.setup_vectorstore(document_paths)
-        results = self.retriever.invoke(query)
-        if len(results) > 0:
-            # Check the confidence score of the first result
-            # (assuming retriever returns sorted results)
-            top_result = results[0]
-            score = top_result.metadata.get("score", 0.0)
-    
-            if score >= threshold:
-                return top_result, score
-        return None, 0.0
-    
+        coding_keywords = [
+            'code', 'Python', 'bug', 'error', 'exception', 'syntax', 'Pygame', 'debug', 
+            'program', 'programming', 'variable', 'loop', 'if', 'function', 'print', 
+            'input', 'string', 'number', 'list', 'tuple', 'dictionary', 'module', 
+            'import', 'turtle', 'indentation', 'syntax', 'Pygame', 'math', 'shape', 
+            'for', 'while', 'repeat', 'class', 'object', 'method', 'attribute', 
+            'operator', 'condition', 'true', 'false', 'boolean', 'integer', 'float', 
+            'type', 'index', 'range', 'len', 'append', 'pop', 'remove', 'sort', 
+            'reverse', 'random', 'sleep', 'time', 'pygame', 'GTK', 'logic', 'numbers', 'correct', 'answer'
+        ]
+        return any(keyword.lower() in question.lower() for keyword in coding_keywords)
+
+
```

Comment:
We didn't agree on this change, why is it here?

Reply:
Ok then I'll remove this out from our code.